library/std/src/os/unix/net/listener.rs

```diff
@@ -262,6 +262,30 @@ impl IntoRawFd for UnixListener {
     }
 }
 
+#[unstable(feature = "io_safety", issue = "87074")]
+impl AsFd for UnixListener {
+    #[inline]
+    fn as_fd(&self) -> BorrowedFd<'_> {
+        self.0.as_inner().as_fd()
+    }
+}
+
+#[unstable(feature = "io_safety", issue = "87074")]
+impl From<OwnedFd> for UnixListener {
+    #[inline]
+    fn from(fd: OwnedFd) -> UnixListener {
+        UnixListener(Socket::from_inner(FromInner::from_inner(OwnedFd::from(fd))))
```

Comment:
Unless I'm missing something, you shouldn't need `OwnedFd::from` here; I think that's just an identity, since `fd` is already an `OwnedFd`.